VerifyReplication/index.js

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable no-console */
+
+const { Logger } = require('werelogs');
+
+const { defaults, mandatoryVars } = require('./constants');
+const { verifyReplication } = require('./verifyReplication');
+
+const log = new Logger('s3utils:verifyReplication');
+const status = {
+    srcListedCount: 0,
+    dstProcessedCount: 0,
+    missingInDstCount: 0,
+    sizeMismatchCount: 0,
+    replicatedCount: 0,
+    dstBucketInProgress: null,
+    srcBucketInProgress: null,
+};
+const {
+    SRC_ENDPOINT,
+    SRC_BUCKET,
+    SRC_ACCESS_KEY,
+    SRC_SECRET_KEY,
+    DST_ENDPOINT,
+    DST_ACCESS_KEY,
+    DST_SECRET_KEY,
+    DST_BUCKET,
+    LOG_PROGRESS_INTERVAL,
+    SRC_BUCKET_PREFIXES,
+    LISTING_LIMIT,
+    LISTING_WORKERS,
+    BUCKET_MATCH,
+    COMPARE_OBJECT_SIZE,
+    COMPARE_OBJECT_ALL_VERSIONS,
+    DST_STORAGE_TYPE,
+    DST_REGION,
+    DST_MD_REQUEST_WORKERS,
+    HTTPS_CA_PATH,
+    HTTPS_NO_VERIFY,
+} = process.env;
+
+const USAGE = `
+VerifyReplication/index.js
+
+This script verifies replication by comparing source and destination objects.
+
+Usage:
+    node VerifyReplication/index.js
+
+Mandatory environment variables:
```

Comment:
I do see an improvement that might be useful for large buckets, in case the tool stops for any reason: why don't we periodically store the progress of workers, so that we can either start or restart at any point from a specific object in the original bucket?
What do you think?

Reply:
@williamlardier yes its on my to-do list, it would certainly be an improvement. The current iteration is for a preview (not final).

Reply:
It is adding more complexity than originally visioned. It can be implemented as an improvement, if needed.